### DISCUSSION TITLE
Idea of preset

### DISCUSSION THREAD

**Discussion by ghost** [unocss/unocss#1194] [Q&A] [answered]
Hi, thanks for the amazing work. I was using TailwindCSS in the past and I wanna try UnoCSS in my current and future projects.

I'm playing around with the @unocss/runtime. However, I have some questions:

1) The Mini CDN Builder comes with @unocss/preset-mini and @unocss/preset-attributify preset, but there is no @unocss/preset-mini preset alone?
2) When I use the Mini of CDN Build, it generates some unused CSS properties on the frontend and some empty variables(--un-empty) as you can see in the screenshot below. I do not quite understand the idea of preset, is it different from TailwindCSS where it will create a cache version that only necessary codes are output to the frontend?

![screenshot](https://user-images.githubusercontent.com/73235650/176630436-fa9629e1-48f4-4ce0-b133-95c160bb2e15.png)

3) What is the use of the Core CDN Build? Do we need to create our custom preset? If so, how do we create our own preset?

**Comment by antfu** [comment, marked as answer]
1. You can always use build tools to bundle the runtime with your own config. It's not possible for us to provide every combination. See https://github.com/unocss/unocss/tree/main/packages/runtime#bundler-usage
2. It's the same behavior as the latest Tailwind (in it's preflight)
